Hardhat Overhaul (#1306)
moonbeam-foundation/moonbeam-docs: builders/ethereum/dev-env/hardhat.md

3. Install Hardhat.

    ```sh
    npm install --save-dev hardhat
    ```

4. Create a Hardhat project.

    ```sh

>     npx hardhat init

Reviewer: in v3 the command actually is suppoesd to be `npx hardhat --init` per their docs, I only know this cause I just made this change in Polkadot haha

https://hardhat.org/docs/getting-started#installation

The Hardhat configuration file is the entry point into your Hardhat project. It defines various settings and options for your Hardhat project, such as the Solidity compiler version to use and the networks you can deploy your contracts to.

> Hardhat 3 is ESM-first. If you stay with JavaScript, keep `"type": "module"` in your `package.json`. A minimal `hardhat.config.js` looks like:

Reviewer: ```suggestion
If you are using JavaScript, keep `"type": "module"` in your `package.json`. A minimal `hardhat.config.js` looks like:
```

```js
import { defineConfig } from "hardhat/config";

export default defineConfig({
  solidity: "0.8.28",

> });

Reviewer: ```suggestion
import { defineConfig } from 'hardhat/config';

export default defineConfig({
  solidity: '0.8.28',
});
```

    npx hardhat keystore set DEV_PRIVATE_KEY
    ```

!!! warning
    The Hardhat console task does not currently prompt for keystore secrets. Use environment variables for your config variables before running `npx hardhat console`, or interact through scripts/tasks instead of the console when using the keystore.

Then, update your configuration file to use the encrypted secrets and ESM syntax:

=== "Moonbeam"

    ```js

>     import hardhatEthers from "@nomicfoundation/hardhat-ethers";

Reviewer: can you pls update these snippets to use single quotes?

There are some limitations to be aware of when using Hardhat forking. You cannot interact with any of the Moonbeam precompiled contracts or their functions. Precompiles are a part of the Substrate implementation and therefore cannot be replicated in the simulated EVM environment. This prohibits you from interacting with cross-chain assets on Moonbeam and Substrate-based functionality such as staking and governance.

> Earlier versions of Hardhat required a manual patch to fork Moonbeam. Hardhat 3 includes the fix, so no manual patching is needed.

Reviewer: ```suggestion
```

I don't think we even need to mention this anymore. Can just scrap this section
